homework/insensitive-palindrom/insensitivePalindrom.cpp

```diff
@@ -0,0 +1,29 @@
+#include <string>
+#include <cctype>
+#include <iostream>
+#include <algorithm>
+
+std::string& sentenceToLower(std::string& str)
+{
+    for(auto & s: str)
+        s = std::tolower(s);
+    return str;
+}
+
+std::string& onlyAlphanumeric(std::string& str)
+{
+    std::string alphanum;
+    std::for_each(str.begin(), str.end(), 
+                [&alphanum](auto & s){ if( std::isalnum(s)) alphanum.push_back(s);});
+    str = alphanum;
```

Comment:
Maybe in this case will be better use algorithm copy_if or remove_if?
You can also merge onlyAlphanumeric() and sentenceToLower() in one algorithm